### PR TITLE
Show/hide reconciled transactions in the account view

### PR DESCRIPTION
**Pull request by davidkus**
Adds toggle to show/hide reconciled transactions in the account view.
This functions similar to how it works in YNAB:
1. Setting is stored in local prefs (persisted between refresh)
2. Setting is per-account

edit:
Just some more context on this:

I had previously added a filter for reconciled transactions in #2108. But this doesn't really work as I'd want it (and other people have requested). Really what we want is for this to persist, and behave like the YNAB "view" for hiding reconciled transactions (e.g. still show scheduled transactions).

https://github.com/actualbudget/actual/assets/5488094/72b60afd-51d4-49f8-b8d3-ee9618de8ec6

### CODE REVIEW

**Comment by youngcw** [comment]
This is looking pretty good.  I noticed that if you have locked hidden, then add a filter, then show locked transactions again, that the filter no longer applies correctly.

**Comment by davidkus** [comment]
> This is looking pretty good. I noticed that if you have locked hidden, then add a filter, then show locked transactions again, that the filter no longer applies correctly.

Good catch, should be fixed with latest commit.

File: packages/desktop-client/src/components/accounts/Account.jsx [hunk 1]
```diff
@@ -651,6 +658,23 @@ class AccountInternal extends PureComponent {
           this.setState({ showCleared: true });
         }
         break;
+      case 'toggle-reconciled':
+        const refilterTransactions = () => {
+          const filters = this.state.filters;
+          if (filters.length > 0) {
+            this.applyFilters([...filters]);
+          } else {
+            this.fetchTransactions();
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/accounts/Account.jsx:667]: I think `fetchTransactions` accepts a `filters` parameter so you can just probably pass in the filters to that method and remove the if else block here. :)

Comment (davidkus) [reply packages/desktop-client/src/components/accounts/Account.jsx]: Updated

**Review by joel-jeremy — APPROVED** [review]
LGTM!

**Pull request merged**